library/src/test/java/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveDataTests.kt

```diff
@@ -0,0 +1,122 @@
+package com.getstream.sdk.chat.viewmodel.messages
+
+import androidx.arch.core.executor.testing.InstantExecutorExtension
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.getstream.sdk.chat.adapter.MessageListItem
+import com.getstream.sdk.chat.createMessage
+import com.getstream.sdk.chat.view.messages.MessageListItemWrapper
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldNotBeEqualTo
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(InstantExecutorExtension::class)
+class MessageListItemLiveDataTests {
+
+    private val currentUser = User()
+    private lateinit var messages: MutableLiveData<List<Message>>
+    private lateinit var threadMessages: MutableLiveData<List<Message>>
+    private lateinit var typing: MutableLiveData<List<User>>
+    private lateinit var reads: MutableLiveData<List<ChannelUserRead>>
+
+    private lateinit var observer: TestObserver<MessageListItemWrapper>
+
+    private lateinit var sut: MessageListItemLiveData
+
+    @BeforeEach
+    fun setUp() {
+        observer = TestObserver()
+        messages = MutableLiveData(listOf())
+        threadMessages = MutableLiveData(listOf())
+        typing = MutableLiveData(listOf())
+        reads = MutableLiveData(listOf())
+
+        sut = MessageListItemLiveData(currentUser, messages, threadMessages, typing, reads)
+    }
+
+    @Test
+    fun `When messages are changed if live data in the thread state should not be observed any values `() {
+        sut.observeForever(observer)
+        threadMessages.value = listOf(createMessage())
+        observer.cleanState()
+
+        messages.value = listOf(createMessage())
+
+        observer.lastObservedValue shouldBeEqualTo null
+    }
+
+    @Test
+    fun `When messages are changed if live data is not in the thread state should observe last assigned value to messages`() {
+        sut.observeForever(observer)
+        val sendingMessage = createMessage()
+
+        messages.value = listOf(sendingMessage)
+
+        observer.lastObservedValue.also { result ->
+            result shouldNotBeEqualTo null
+            result!!.listEntities.any { item ->
```

Comment:
Kluent has a `shouldNotBeNull` method we could use on the first line here, and then we wouldn't need to use `!!` later on, because it has [contract support](https://github.com/MarkusAmshove/Kluent/blob/e94cbd8dd1a4a7a5cc3271f937840bbf1b380974/common/src/main/kotlin/org/amshove/kluent/Basic.kt#L26-L33).

Reply:
Thanks for the comment, I hasn't worked with fluent before